### PR TITLE
Add Dynamic Method Call to JSON Editor

### PR DESCRIPTION
**Pull request by natankeddem**
This PR adds the ability call dynamic methods on the JSON Editor.

### CODE REVIEW

File: nicegui/elements/json_editor.py [hunk 1]
```diff
@@ -42,3 +43,7 @@ def properties(self) -> Dict:
     def update(self) -> None:
         super().update()
         self.run_method('update_editor')
+
+    def call_editor_method(self, name: str, arg: Optional[str] = None, timeout: float = 1,
```
Comment (rodja) [inline nicegui/elements/json_editor.py:47]: I would suggest to name the function `run_api_method`. We already provide a `run_method` function from the base class `Element` and the naming should be consistent. In AgGrid we have named it `call_api_method` which we should also change in the next major release.

Comment (natankeddem) [reply nicegui/elements/json_editor.py]: done

**Review by falkoschindler — COMMENTED** [review]
Thanks for the pull request, @natankeddem!

I just reviewed it and did some changes:
- I simplified the JavaScript function to accept an arbitrary number of arguments.
- After a discussion with @rodja we agreed on the name `run_editor_method`. As part of this PR we'll adjust AG Grid's `call_api_method` and `call_column_api_method` accordingly.
- I added a docstring and split the demo into two, so that both remain neatly short.

The PR is basically ready to merge. But I've got one final concern regarding the API:

The `run_editor_method` expects the argument(s) to be JavaScript strings. That's ok, but is important to know since
```py
editor.run_editor_method('updateProps', {'readOnly': True})
```
won't work. We could automatically JSON-serialize other data types, like convert the dict `{'readOnly': True}` into `'{readOnly: true}'` - but we wouldn't know whether to convert a _string_ or not. In other places we indicate dynamic properties using a key with a leading colon ":". But here we don't necessarily have keys.

Maybe I'm thinking too complicated. But an API like this is hard to change, once it's out in the world. Therefore I'd like to think about it once again before merging into main. Do you have ideas how we could handle such arguments?

**Comment by natankeddem** [comment]
Following the convention you have already established, we could do the following. If the method name is prefixed by `:` the arguments would not be serialized and if it is there the data will be serialized. The following 2 calls should work and accomplish the same thing:
```py
editor.run_editor_method('updateProps', {'readOnly': True})
editor.run_editor_method(':updateProps', "{'readOnly': true}")
```

**Comment by falkoschindler** [comment]
Even though this scenario is a little different from "dynamic props", I like the idea of adding ":" to the method name. So you could write
```py
editor.run_editor_method('updateProps', {'readOnly': True})
```
where the dictionary will be used as is, and
```py
editor.run_editor_method(':expand', 'path => true')
```
where the filter function will be evaluated on the client.

@rodja I'd like to hear your opinion about this suggestion. Is it too cryptic? Or just right because we consistently use ":" to indicate dynamic attributes.

**Comment by natankeddem** [comment]
This might be worth a discussion poll?

**Comment by rodja** [comment]
The alternative to the colon would be a parameter, right?

```py
editor.run_editor_method('expand', 'path => true', evaluate_on_client=True)
```

I think the colon is "nicer" and more consistent. Still, a poll would be interesting... I'm curious how big the community is for such a feature.

**Comment by natankeddem** [comment]
Another alternative would be to only serialize the argument if it is anything but a string. I think this is safe but I'm not if there are some edgecases I am not accounting for.

**Comment by falkoschindler** [comment]
@natankeddem Yes, that sounds compelling. But although I don't see an edge case with this specific library, in general there could be something like `editor.run_editor_method('set_background_color', 'red')`. And because `'red'` is a string, we would evaluate it on the client to `red`, which is undefined. So some strings need to be evaluated, but not all of them.

**Comment by falkoschindler** [comment]
@rodja @natankeddem Feel free to create a poll on this subject. Personally I tend towards Natan's proposal `run_editor_method(':updateProps', ...)` and would like to close this PR within the next few days. But I think we could also change this API later, since a parameter `evaluate_on_client` (or similar) wouldn't conflict with the colon prefix.

**Review by falkoschindler — APPROVED** [review]
Ok I introduced the colon prefix ":" for the method name. I really like this solution.

I guess this PR is ready to merge. Thanks again for your contribution, @natankeddem! 😊

**Comment by natankeddem** [comment]
Thank you @falkoschindler! Sorry I didn't get around to the poll. I think this is a good solution which maintains continuity with the existing mechanisms in NiceGUI.

**Pull request merged**